aten/src/ATen/core/type.cpp

```diff
@@ -319,7 +319,31 @@ c10::optional<TypePtr> unifyTypes(const TypePtr& t1, const TypePtr& t2) {
   return c10::nullopt;
 }
 
-MatchTypeReturn matchTypeVariables(TypePtr formal, TypePtr actual, TypeEnv& type_env) {
+c10::optional<TypePtr> unifyTypeList(at::ArrayRef<TypePtr> elements) {
```

Comment:
Why does this return optional<TypePtr> but unifyTypes return TypePtr that might be null. This should be consistent with TypePtr.

Reply:
`unifyTypes` returns an optional